### ISSUE TITLE
Unable to attach VS code to container on remote server

### ISSUE THREAD

**Issue by acaala** [microsoft/vscode-remote-release#5074]
Issue Type: <b>Bug</b>

When trying to attach vs code to a container on my remote server it creates an alert saying Docker version 17.12.0 or later is required.

using the `docker  -v` on a terminal i get 
`Docker version 20.10.6, build 370c289`

![Screenshot 2021-05-19 at 11 43 55](https://user-images.githubusercontent.com/66971974/118800238-98ea5a80-b897-11eb-88a4-d38d73158d52.png)
![Screenshot 2021-05-19 at 11 44 03](https://user-images.githubusercontent.com/66971974/118800243-9a1b8780-b897-11eb-90d0-32cd5ce74441.png)

clicking install gives me this

![Screenshot 2021-05-19 at 11 50 42](https://user-images.githubusercontent.com/66971974/118801169-9dfbd980-b898-11eb-8076-15568fb53569.png)

![Screenshot 2021-05-19 at 11 52 21](https://user-images.githubusercontent.com/66971974/118801217-ace28c00-b898-11eb-95ed-469873492023.png)

Extension version: 0.177.2
VS Code version: Code 1.56.2 (054a9295330880ed74ceaedda236253b4f39a335, 2021-05-12T17:44:30.902Z)
OS version: Darwin x64 20.3.0
Remote OS version: Linux x64 5.4.0-73-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz (4 x 1800)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 4, 3|
|Memory (System)|8.00GB (0.02GB free)|
|Process Argv|--crash-reporter-id 801c304a-f311-4e66-9f5e-3803c37d94b6|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: 109.228.51.253|
|OS|Linux x64 5.4.0-73-generic|
|CPUs|Intel(R) Xeon(R) Gold 5120 CPU @ 2.20GHz (2 x 2194)|
|Memory (System)|1.91GB (0.16GB free)|
|VM|25%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspyt653:30270858
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
vspre833:30267464
pythonptprofiler:30281270
vsdfh931:30280409
vshan820:30294714
pythondataviewer:30285071
vscus158cf:30286554
vscgsv2:30301612
vscorehovct:30302760
bridgeflightcf:30302070
vscod805cf:30301675

```

</details>

**Comment by eandersons** [comment]
@acaala, since you closed this issue, I assume that you found the reason for this alert. Did you manage to resolve the underlying issue?
I am asking, because I am getting the same alert when I try to attach VS Code from Linux machine to a container on a remote Mac.
I can attach to container's shell.

**Comment by acaala** [comment]
Yeah I needed docker installed on my local machine.

I assumed as it was installed on my remote server I didn't need it on my local machine. 

Hopefully this will help.

**Comment by eandersons** [comment]
I have Docker installed on both local and remote machines.

Below are my findings in case other fellow containerists find them useful.

I tried to figure something out for a while, and then I found the following in VS Code logs (`/home/{user}/.config/Code/logs/{yyyyddmmThhmmss}/exthost1/ms-vscode-remote.remote-containers/remoteContainers-{yyyy-mm-ddThh-mm-ss.000Z.log`):

```text
[2021-06-18T23:14:21.663Z] Start: Run: docker version --format {{.Server.APIVersion}}
[2021-06-18T23:14:21.663Z] Remote-Containers 0.183.0 in VS Code 1.57.1 (507ce72a4466fbb27b715c3722558bb15afa9f48).
[2021-06-18T23:14:22.068Z] Stop (405 ms): Run: docker version --format {{.Server.APIVersion}}
[2021-06-18T23:14:22.072Z] error during connect: Get "http://docker/v1.24/version": command [ssh -- {user}@{host} docker system dial-stdio] has exited with exit status 127, please make sure the URL is valid, and Docker 18.09 or later is installed on the remote host: stderr=zsh:1: command not found: docker
```

At least that was something to work on and that lead to <https://github.com/docker/for-mac/issues/4382#issuecomment-603031242> that helped me to solve the alert "Docker version 17.12.0 or later required." and the underlaying error `zsh:1: command not found: docker` with the following actions on remote machine:

- edit `/etc/ssh/sshd_config` to change `#PermitUserEnvironment no` to `PermitUserEnvironment yes`;
- restart `sshd` to apply configuration changes: `sudo killall sshd` or disable and reenable "Remote Login" in System Preferences > Sharing;
- create `~/.ssh/environment` that contains:
  
  ```shell
  PATH=/usr/local/bin:$PATH
  ```

  in my case `sshd` was compiled with `PATH=/usr/bin:/bin:/usr/sbin:/sbin` therefore the Docker binary could not be found when VS Code executed the command `ssh -- {user}@{host} docker system dial-stdio`.
